osu.Server.Spectator/Hubs/MultiplayerHub.cs

```diff
@@ -58,49 +58,97 @@ public async Task<MultiplayerRoom> JoinRoom(long roomId)
                 // add the user to the room.
                 var roomUser = new MultiplayerRoomUser(CurrentContextUserId);
 
-                MultiplayerRoom room;
+                // track whether a new room was fetched from the database and assigned to a usage.
+                bool newRoomFetched = false;
```

Comment:
This comment... isn't all that accurate. If the flag were to be set to `true` *after* `RetrieveRoom()`, then it would be, but it isn't. In fact the cleanup code even corrects for that by checking null room (which I also find a bit non-straightforward).

Reply:
I repurposed this halfway through, will rename the variabl----